src/libcrun/container.c

```diff
@@ -3706,6 +3708,148 @@ libcrun_container_update_from_values (libcrun_context_t *context, const char *id
   return ret;
 }
 
+void
+populate_array_field (const char ***field, const char *array[], size_t num_elements)
+{
+  *field = xmalloc0 ((num_elements + 1) * sizeof (char *));
+  size_t i;
+  for (i = 0; i < num_elements; i++)
+    {
+      (*field)[i] = xstrdup (array[i]);
+    }
+
+  (*field)[i] = NULL; // Terminate the array with NULL
+}
+
+void
+populate_capabilities (struct features_info_s ***info, const char ***capabilities, size_t *num_capabilities)
```

Comment:
can't `struct features_info_s ***info` just be `struct features_info_s *info`?

Reply:
Yup, fixed that